src/Symfony/Component/Translation/Extractor/BlankableExtractorInterface.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Extractor;
+
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * Extracts translation messages from a directory or files to the catalogue.
+ * New found messages are injected to the catalogue using the prefix.
+ * New found message can be extracted as blank strings.
+ *
+ * @author Michel Salib <michelsalib@hotmail.com>
+ */
+interface BlankableExtractorInterface
+{
+    /**
+     * Extracts translation messages from files, a file or a directory to the catalogue.
+     *
+     * @param string|iterable<string> $resource Files, a file or a directory
+     */
+    public function extract(string|iterable $resource, MessageCatalogue $catalogue);
+
+    /**
+     * Sets the prefix that should be used for new found messages.
+     */
+    public function setPrefix(string $prefix);
+
+    /**
+     * Extract new-found messages as blank strings.
+     */
+    public function blank(bool $blank);
```

Comment:
i tend to prefer a new virtual method on existing ExtractorInterface (`@method` annotation)

Reply:
Using the existing interface would make the change much smaller and easier. I didn't know that was possible from my understanding of the BC guidelines.

So your suggestion would basically be this and removing the new `BlankableExtractorInterface`? (plus the resulting follow-up changes of course)


```
diff --git a/src/Symfony/Component/Translation/Extractor/ExtractorInterface.php b/src/Symfony/Component/Translation/Extractor/ExtractorInterface.php
index b76a7f20ba..e8745c6d15 100644
--- a/src/Symfony/Component/Translation/Extractor/ExtractorInterface.php
+++ b/src/Symfony/Component/Translation/Extractor/ExtractorInterface.php
@@ -18,6 +18,8 @@ use Symfony\Component\Translation\MessageCatalogue;
  * New found messages are injected to the catalogue using the prefix.
  *
  * @author Michel Salib <michelsalib@hotmail.com>
+ *
+ * @method blank(bool $blank)
  */
 interface ExtractorInterface
 {

```

Reply:
for this design it's the fastest route yes (i think it's sensible design :+1: )

Reply:
I made the changes in a separate commit. Like this approach much better 👍